### ISSUE TITLE
[BUG] Inconsistent Behavior between Collector and `EnvBase.rollout`

### ISSUE THREAD

**Issue by btx0424** [pytorch/rl#1569]
## Describe the bug

Currently the rollout logic of `Collector` and `EnvBase.rollout` differs in many ways. One problem is that, after the refactor which added support for complex `done_keys`, `Env.rollout` currently runs into `DONE_AFTER_RESET_ERROR` on `TransformedEnv`.

## To Reproduce

The following code

```python
from torchrl.envs import GymEnv
from torchrl.envs.transforms import TransformedEnv, InitTracker, Compose

env = GymEnv("Pendulum-v1")
env = TransformedEnv(env, Compose(InitTracker()))

rollout = env.rollout(2000, break_when_any_done=False)
print(rollout)
```

would result in 

```
...
  File ".../rl/torchrl/envs/common.py", line 1321, in reset
    raise DONE_AFTER_RESET_ERROR
RuntimeError: Env was done after reset on specified '_reset' dimensions. This is (currently) not allowed.
```

## Reason and Possible fixes

This is because in `TransformedEnv._reset`:

```python
    def _reset(self, tensordict: Optional[TensorDictBase] = None, **kwargs):
        if tensordict is not None:
            tensordict = tensordict.clone(recurse=False)
        out_tensordict = self.base_env._reset(tensordict=tensordict, **kwargs)
        if tensordict is not None:
            out_tensordict = tensordict.update(out_tensordict) # this would retain the dones
        out_tensordict = self.transform.reset(out_tensordict)

        mt_mode = self.transform.missing_tolerance
        self.set_missing_tolerance(True)
        out_tensordict = self.transform._call(out_tensordict)
        self.set_missing_tolerance(mt_mode)
        return out_tensordict
```

But in `Collectors.rollout`, the error won't occur because only "_reset" is passed in. This issue is also related to #1465 . 
I think we should decide the desired behavior and make the logic more consistent. Maybe consider resetting the dones in `EnvBase.reset`? 

## Checklist

- [x] I have checked that there is no similar issue in the repo (**required**)
- [x] I have read the [documentation](https://github.com/pytorch/rl/tree/main/docs/) (**required**)
- [x] I have provided a minimal working example to reproduce the bug (**required**)

**Comment by vmoens** [comment]
Thanks for this @btx0424!
I agree that this is far from optimal.
The bug you encountered is solved in #1539 but this PR needs some more work. Feel free to join the discussion to give your input! See you there :)

**Comment by vmoens** [comment]
I believe #1539 fixed it!
